src/CopyOnWrite/Copy.cs

```diff
@@ -1010,38 +1009,17 @@ private bool TryCopyOnWrite(string source, string dest)
         /// <returns>True when CopyOnWrite appears to be supported.</returns>
         private bool IsCopyOnWriteSupported(string source, string dest)
         {
-            if (source.StartsWith(@"\\", StringComparison.OrdinalIgnoreCase) || dest.StartsWith(@"\\"))
+            if (DisableCoW)
             {
                 return false;
             }
 
-            var sourceDrive = Path.GetPathRoot(source);
-            var destDrive = Path.GetPathRoot(dest);
-
-            if (!sourceDrive.Equals(destDrive, StringComparison.OrdinalIgnoreCase))
+            if (source.StartsWith(@"\\", StringComparison.Ordinal) || dest.StartsWith(@"\\", StringComparison.Ordinal))
             {
                 return false;
             }
 
-            try
-            {
-                return _reFsDrives.GetOrAdd(
-                    sourceDrive,
-                    (key) => new Lazy<bool>(() =>
-                    {
-                        var supportsCloneFile = _cow.CopyOnWriteLinkSupportedInDirectoryTree(key);
-                        Log.LogMessage(MessageImportance.Low,
-                            supportsCloneFile
-                                ? $"Drive {key} has support for CloneFile. All copies will attempt to use CloneFile."
-                                : $"Drive {key} does not have support for CloneFile. File.Copy will be used.");
-                        return supportsCloneFile;
-                    })).Value;
-            }
-            catch (Exception ex)
```

Comment:
The try/catch was added to fix #453. Should probably just leave it?

Reply:
Will restore. Interesting case in the bug, will add that to the CoW library as well

Reply:
Added that case to the CoW package in 0.3.4. (0.3.5 in latest commit sets the AssemblyVersion to 0.0.0.1 to avoid collisions if a repo includes differing versions of Microsoft.Build.CopyOnWrite and Microsoft.Build.Artifacts.)